### PR TITLE
Approach #2. Mapping constraint names to readable format

### PR DESCRIPTION
**Pull request by pecop2**
## Description

Maps used to map certain operations to easily understandable strings. The display_names are fully generated from the constraint value, field, operation etc. Previously created simple "name" property is still existent.

**This is the second approach, different than the one initially proposed**

### General Checklist

* [ ] Tests added for this feature/bug
      if it was a bug, test must cover it.
* [ ] Conform by the style guides, by using formatter
* [ ] Documentation updated
* [ ] (optional) Please add a label to your PR

### CODE REVIEW

File: src/whylogs/core/statistics/constraints.py [hunk 1]
```diff
@@ -196,6 +232,18 @@ def _matches_json_schema(json_data: Union[str, dict], json_schema: Union[str, di
     Op.SUM: lambda v: sum(v),
 }
 
+_multi_column_value_funcs_names = {
+    Op.LT: "ess than",
```
Comment (richard-rogers) [inline src/whylogs/core/statistics/constraints.py:236]: less than?

File: src/whylogs/core/statistics/constraints.py [hunk 2]
```diff
@@ -967,6 +1085,10 @@ def report(self) -> List[tuple]:
             return v
         return None
 
+    def display_report(self, feature_name: str) -> List[tuple]:
```
Comment (richard-rogers) [inline src/whylogs/core/statistics/constraints.py:1088]: Looks like return type should be `Optional[List[tuple]]`

File: src/whylogs/core/statistics/constraints.py [hunk 3]
```diff
@@ -1013,14 +1135,44 @@ def __init__(
...
+            dependent_cols = f"{Op.Name(self.internal_op)} {dependent_cols}"
+        val_or_ref_columns = self._get_value_or_ref_columns()
+
         return self._name if self._name is not None else f"multi column value {dependent_cols} {Op.Name(self.op)} {val_or_ref_columns}"
 
+    def display_name(self):
+        if not hasattr(self, "_display_name_dict"):
+            self._set_display_name_dictionary()
+
+        cols = "Column" if len(self._display_name_dict["dependent_columns"]) == 1 else "Columns"
```
Comment (richard-rogers) [inline src/whylogs/core/statistics/constraints.py:1148]: Will we need to localize this?

File: src/whylogs/core/statistics/constraints.py [hunk 4]
```diff
@@ -1172,16 +1331,35 @@ def to_protobuf(self) -> MultiColumnValueConstraintMsg:
...
     @staticmethod
     def from_protobuf(msg: ValueConstraintMsgs) -> "MultiColumnValueConstraints":
-        value_constraints = [MultiColumnValueConstraint.from_protobuf(c) for c in msg.constraints]
+        value_constraints = [MultiColumnValueConstraint.from_protobuf(c) for c in msg.multi_column_constraints]
         if len(value_constraints) > 0:
-            return MultiColumnValueConstraints({v.name: v for v in value_constraints})
+            return MultiColumnValueConstraints(value_constraints)
+        return None
+
+    def to_protobuf(self) -> ValueConstraintMsgs:
```
Comment (richard-rogers) [inline src/whylogs/core/statistics/constraints.py:1349]: Looks like return type should be `Optional[ValueConstraintMsgs]`

File: src/whylogs/core/statistics/constraints.py [hunk 5]
```diff
@@ -1172,16 +1331,35 @@ def to_protobuf(self) -> MultiColumnValueConstraintMsg:
...
+        v = [c.to_protobuf() for c in self.raw_value_constraints.values()]
+        v.extend([c.to_protobuf() for c in self.coerced_type_constraints.values()])
+        if len(v) > 0:
+            vcmsg = ValueConstraintMsgs()
+            vcmsg.multi_column_constraints.extend(v)
+            return vcmsg
+        return None
+
+    def display_report(self) -> List[tuple]:
```
Comment (richard-rogers) [inline src/whylogs/core/statistics/constraints.py:1358]: `Optional[List[tuple]]`

File: tests/unit/core/statistics/test_constraints.py [hunk 6]
```diff
@@ -914,9 +915,21 @@ def test_apply_func_value_constraints(local_config_path):
     )
 
 
+def test_apply_func_invalid_init():
+
+    with pytest.raises(ValueError):
```
Comment (richard-rogers) [inline tests/unit/core/statistics/test_constraints.py:920]: parametrize

File: tests/unit/core/statistics/test_constraints.py [hunk 7]
```diff
@@ -2772,3 +2823,120 @@ def test_column_values_unique_within_row_constraint_invalid_params():
...
+    assert sum_of_values["name"] == f"multi column value SUM {col_set} {Op.Name(Op.EQ)} 100"
+    assert sum_of_values["dependentColumns"] == col_set
+    assert sum_of_values["op"] == Op.Name(Op.EQ)
+    assert pytest.approx(sum_of_values["value"], 0.01) == 100
+    assert sum_of_values["internalDependentColumnsOp"] == Op.Name(Op.SUM)
+    assert sum_of_values["verbose"] is False
+
+
+def test_display_name():
+    cvisc = columnValuesInSetConstraint(value_set={2, 5, 8})
```
Comment (richard-rogers) [inline tests/unit/core/statistics/test_constraints.py:2868]: Can we break this up into smaller functions?

**Review by richard-rogers — CHANGES_REQUESTED** [review]

**Comment by pecop2** [comment]
Since we already continued with the other approach, should this PR be closed ? @jamie256

**Comment by jamie256** [comment]
We implemented a different approach, closing this PR.

**Pull request closed without merging**